### PR TITLE
add lint and pre-commit

### LINKED ISSUE #100: add flake8, black, isort, pydocstring  .... or use yapf
lets the war begin @kylebarron! the first one to submit a PR wins!

### PR DESCRIPTION
**Pull request by vincentsarago**
closes #100

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+repos:
```
Comment (vincentsarago) [inline .pre-commit-config.yaml:1]: for rio-tiler 🤷

File: README.md [hunk 2]
```diff
@@ -1,4 +1,8 @@
-# aiocogeo [![CircleCI](https://circleci.com/gh/geospatial-jeff/aiocogeo/tree/master.svg?style=svg)](https://circleci.com/gh/geospatial-jeff/aiocogeo/tree/master)[![codecov](https://codecov.io/gh/geospatial-jeff/aiocogeo/branch/master/graph/badge.svg)](https://codecov.io/gh/geospatial-jeff/aiocogeo)
+# aiocogeo
+[![GHA](https://github.com/geospatial-jeff/aiocogeo/workflows/CI/badge.svg)](https://github.com/geospatial-jeff/aiocogeo/actions?query=workflow%3ACI)
+[![codecov](https://codecov.io/gh/geospatial-jeff/aiocogeo/branch/master/graph/badge.svg)](https://codecov.io/gh/geospatial-jeff/aiocogeo)
+[![pypi](https://img.shields.io/pypi/v/aiocogeo?color=%2334D058&label=pypi%20package)](https://pypi.org/project/aiocogeo)
+[![license](https://img.shields.io/github/license/geospatial-jeff/aiocogeo.svg)](https://github.com/geospatial-jeff/aiocogeo/blob/master/LICENSE)
```
Comment (vincentsarago) [inline README.md:5]: ✨

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -37,6 +37,11 @@ jobs:
           AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
         run: tox -e py
 
+      # Run pre-commit (only for python-3.7)
```
Comment (geospatial-jeff) [inline .github/workflows/ci.yml:40]: I think pre-commit should run before tests?

**Comment by geospatial-jeff** [comment]
Closing in favor of #105 

Thanks as always @vincentsarago !

**Pull request closed without merging**